browser/src/control/Control.NavigatorPanel.ts

```diff
@@ -0,0 +1,250 @@
+/* -*- js-indent-level: 8 -*- */
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+/*
+ * JSDialog.NavigatorPanel
+ */
+
+/* global app */
+class NavigatorPanel extends SidebarBase {
+	navigationPanel: HTMLElement;
+	floatingNavIcon: HTMLElement;
+	presentationControlsWrapper: HTMLElement;
+	navigatorDockWrapper: HTMLElement;
+
+	constructor(map: any, options: SidebarOptions) {
+		super(map, options, SidebarType.Navigator);
+	}
+
+	onAdd(map: ReturnType<typeof L.map>) {
+		super.onAdd(map);
+		this.map.on('navigator', this.onNavigator, this);
+		this.map.on('doclayerinit', this.onDocLayerInit, this);
+		this.navigationPanel = document.getElementById(`navigation-sidebar`);
+		this.floatingNavIcon = document.getElementById(`navigator-floating-icon`);
+		this.presentationControlsWrapper = this.navigationPanel.querySelector(
+			'#presentation-controls-wrapper',
+		);
+		this.navigatorDockWrapper = this.navigationPanel.querySelector(
+			'#navigator-dock-wrapper',
+		);
+		this.map.on(
+			'zoomend',
+			this.handleFloatingButtonVisibilityOnZoomChange,
+			this,
+		);
+	}
+
+	onRemove() {
+		super.onRemove();
+		this.map.off('navigator');
+		this.map.off('zoomend');
+		this.map.on('doclayerinit');
+	}
+
+	onDocLayerInit() {
+		// for presentation show slide sorter navigation panel by default
+		if (app.map.getDocType() === 'presentation' && !window.mode.isMobile())
+			app.map.sendUnoCommand('.uno:Navigator');
```

Comment:
I believe this is a costly operation (to setup navigator in the core). And it is done on load.

Can we just show the slide sorter (which doesn't have core dependency) without loading navigator in the core?